### PR TITLE
Makes Stacks Garbage Collect

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Makes stacks garbage collect. To my knowledge, they actually did GC fairly well, but they wrecked absolute havoc on cyborgs (the stacks wouldn't respawn and the borg's module would get gunked up until the item was hard deleted). This makes it so they do, in fact, GC with borgs.

Due to how crazy cyborgs are coded, this really couldn't _not_ be snowflaked, sadly.

Also fixes some inconsistency with cyborgs stacks and nanopaste stacks in terms of total number (they're all 6 now).

:cl: Fox McCloud
fix: Fixes medical cyborgs having one less stack than they should have they recharge
fix: Fixes inconsistency in the medical stacks amounts (they should all be 6)
/:cl:

### CODE REVIEW

File: code/game/objects/items/stacks/stack.dm [hunk 1]
```diff
@@ -195,15 +194,18 @@
...
-/obj/item/stack/proc/use(var/used)
+/obj/item/stack/proc/use(used)
 	if(amount < used)
 		return 0
 	amount -= used
 	if(amount < 1) // Just in case a stack's amount ends up fractional somehow
+		if(isrobot(loc))
+			var/mob/living/silicon/robot/R = loc  //Horrifying cyborg snowflake code that allows stacks to GC and cyborgs not to horrendously break
+			if(locate(src) in R.module.modules)
+				R.module.modules -= src
```
Comment (Fox-McCloud) [inline code/game/objects/items/stacks/stack.dm]: This is awful. Just absolutely awful.

File: code/game/objects/items/stacks/stack.dm [hunk 2]
```diff
@@ -195,15 +194,18 @@
...
 		if(usr)
-			usr.unEquip(src, 1)
-		spawn()
```
Comment (Fox-McCloud) [inline code/game/objects/items/stacks/stack.dm]: Why was this here.

File: code/modules/mob/living/silicon/robot/inventory.dm [hunk 3]
```diff
@@ -233,7 +233,7 @@
 
 /mob/living/silicon/robot/unEquip(obj/item/I)
 	if(I == module_active)
-		deselect_module(get_selected_module())
+		uneq_active(I)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/silicon/robot/inventory.dm]: Even when I solved the problem of borgs not being able to regen their stacks, I found that the item was still being referenced by the module slot it was held in; this fixes that.

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 4]
```diff
@@ -54,7 +54,6 @@
 		var/obj/item/stack/S = O
 
 		if(!S)
-			modules -= null
```
Comment (Fox-McCloud) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]: This was a stupidly cheesy way of clearing out nulls from a list **THAT SHOULD HAVE BEEN CLEARED OUT PROPERLY IN THE FIRST PLACE**

File: code/game/objects/items/stacks/stack.dm (hunk 2 above)
Comment (tigercat2000) [reply code/game/objects/items/stacks/stack.dm]: If I had to guess, it's because stack code is from before qdel, so anything calling use() could experience a sudden long ass pause for del to work it's thing

Comment (Fox-McCloud) [reply code/game/objects/items/stacks/stack.dm]: That'd make sense.

File: code/game/objects/items/stacks/nanopaste.dm [hunk 5]
```diff
@@ -5,7 +5,8 @@
 	icon = 'icons/obj/nanopaste.dmi'
 	icon_state = "tube"
 	origin_tech = "materials=2;engineering=3"
-	amount = 10
+	amount = 6
```
Comment (KasparoVy) [inline code/game/objects/items/stacks/nanopaste.dm]: Does this help with garbage collection? 🤔

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 6]
```diff
@@ -124,10 +123,10 @@
 	module_type = "Medical"
 	subsystems = list(/mob/living/silicon/proc/subsystem_crew_monitor)
 	stacktypes = list(
-		/obj/item/stack/medical/bruise_pack/advanced = 5,
-		/obj/item/stack/medical/ointment/advanced = 5,
-		/obj/item/stack/medical/splint = 5,
-		/obj/item/stack/nanopaste = 5
+		/obj/item/stack/medical/bruise_pack/advanced = 6,
```
Comment (KasparoVy) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]: Does this help with garbage collection? 🤔
